Dockerfile

```diff
@@ -1,33 +1,85 @@
-# NOTE(dperny): for some reason, alpine was giving me trouble
+# syntax=docker/dockerfile:1
+
 ARG GO_VERSION=1.18.8
+ARG PROTOC_VERSION=3.11.4
 ARG DEBIAN_FRONTEND=noninteractive
-ARG BASE_DEBIAN_DISTRO="bullseye"
-ARG GOLANG_IMAGE="golang:${GO_VERSION}-${BASE_DEBIAN_DISTRO}"
 
-FROM ${GOLANG_IMAGE}
+FROM --platform=$BUILDPLATFORM golang:${GO_VERSION}-bullseye AS gobase
+ARG DEBIAN_FRONTEND
+RUN apt-get update && apt-get install -y --no-install-recommends git make rsync
+WORKDIR /go/src/github.com/docker/swarmkit
```

Comment:
Should this be `github.com/moby/swarmkit` now? (module is named `moby/swarmkit/v2`)

Reply:
Would need to first update: https://github.com/moby/swarmkit/blob/67d2b1bff9e0a841cf26efd6fa36e9e3676575d1/Protobuild.toml#L37

Reply:
oh! yes, proto; I recall now that I ran into that myself; perhaps something we should add a comment for to explain "why"